lib/WasmReader/WasmBinaryReader.cpp

```diff
@@ -1033,6 +1021,14 @@ const char16* WasmBinaryReader::ReadInlineName(uint32& length, uint32& nameLengt
         AssertMsg(UNREACHED, "We calculated the length before decoding, what happened ?");
         ThrowDecodingError(_u("Error while decoding utf8 string"));
     }
+    for (size_t i = 0; i < decodedLength; ++i)
+    {
+        const char16 c = contents[i];
+        if (!utf8::IsValidWideChar(c) || c == UNICODE_UNKNOWN_CHAR_MARK)
+        {
+            ThrowDecodingError(_u("Invalid UTF-8 encoding"));
```

Comment:
Doing a char16 at a time means only surrogate pairs would be invalid.  utf8::DecodeUnitsIntoAndNullTerminate should not be able to generate anything invalid (although that API should be the test surface for such a claim), but given invalid utf8 input, would produce � FFFD http://www.fileformat.info/info/unicode/char/0fffd/index.htm (which I'm assuming is this `UNICODE_UNKNOWN_CHAR_MARK` being checked for here).